SignalServiceKit/src/Util/DispatchQueue+OWS.swift

```diff
@@ -87,4 +87,14 @@ extension OWSDispatch {
         let qos = DispatchQoS(qosClass: qosClass, relativePriority: 0)
         return DispatchQueue.sharedQueue(at: qos)
     }
+
+    /// Returns a reverse-DNS queue label namespaced by appending the `suffix` to the current executable's bundleId
+    public static func createLabel(_ suffix: String) -> String {
+        let prefix = Bundle.main.bundleIdentifier ?? {
+            owsFailDebug("Missing bundleID")
+            return "UNDEFINED"
+        }()
+
```

Comment:
Nitpick: It feels strange to have it return `UNDEFINED` as if it's trying to recover from an invalid state. I feel like this should either (1) return `org.whispersystems.signal.` instead of "UNDEFINED" (2) fully crash. I think I prefer the second option.

```suggestion
        guard let prefix = Bundle.main.bundleIdentifier else {
            owsFail("Missing bundleID")
        }
```

Reply:
I'm fine with that! Though, to be clear: dispatch queue labels are [optional and really only used to aid in debugging](https://developer.apple.com/documentation/dispatch/1642205-dispatch_queue_create_with_targe). So in the worst case, this would just result in something like the following showing up in your backtrace:

```
Thread 14 name:   Dispatch queue: UNDEFINED.PKPushRegistry
Thread 14 Crashed:
*** Backtrace details ***
```

Reply:
It's also worth noting that Signal isn't really consistent with naming conventions. In the link above, Apple suggests using a reverse-DNS label format to make it easier to differentiate who created the queue. However, in some old crash I have saved locally:
```
Thread 14 name:   Dispatch queue: GRDB Storage.reader.1
Thread 14 Crashed:
0   libsystem_kernel.dylib        	       0x20a232200 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x21a46f1ac pthread_kill + 268
2   libsystem_c.dylib             	       0x1d5029ca0 abort + 180
3   libc++abi.dylib               	       0x21a3b1b8c abort_message + 132
```

Who created this queue? Did it come from Signal? Or some linked framework? Something namespaced like "org.whispersystems.signal.grdbStorage.reader.1" would make it much more clear who the originator is.

Reply:
Sorry to keep commenting but just one more thing I want to note. I don't think it's even possible for an iOS app to run without a bundleId. I think the optionality comes from macOS land. So your suggestion to hard assert is almost certainly safe.